src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/AwsBedrockEnbeddingProvider.java

```diff
@@ -0,0 +1,158 @@
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.EmbeddingProviderConfigStore;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.ProviderConstants;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeAsyncClient;
+import software.amazon.awssdk.services.bedrockruntime.model.BedrockRuntimeException;
+import software.amazon.awssdk.services.bedrockruntime.model.InvokeModelResponse;
+
+/** Provider implementation for AWS Bedrock. To start we support only Titan embedding models. */
+public class AwsBedrockEnbeddingProvider extends EmbeddingProvider {
+
+  private static final String providerId = ProviderConstants.BEDROCK;
+  private static ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+  private static ObjectReader or = new ObjectMapper().reader();
+
+  public AwsBedrockEnbeddingProvider(
+      EmbeddingProviderConfigStore.RequestProperties requestProperties,
+      String baseUrl,
+      String modelName,
+      int dimension,
+      Map<String, Object> vectorizeServiceParameters) {
+    super(
+        requestProperties,
+        baseUrl,
+        modelName,
+        acceptsTitanAIDimensions(modelName) ? dimension : 0,
+        vectorizeServiceParameters);
+  }
+
+  @Override
+  public Uni<Response> vectorize(
+      int batchId,
+      List<String> texts,
+      EmbeddingProvider.Credentials credentials,
+      EmbeddingRequestType embeddingRequestType) {
+    if (!credentials.accessKeyId().isPresent() || !credentials.secretAccessKey().isPresent()) {
+      throw ErrorCode.EMBEDDING_PROVIDER_AUTHENTICATIN_KEYS_NOT_PROVIDED.toApiException();
+    }
+
+    AwsBasicCredentials awsCreds =
+        AwsBasicCredentials.create(
+            credentials.accessKeyId().get(), credentials.secretAccessKey().get());
+
+    BedrockRuntimeAsyncClient client =
+        BedrockRuntimeAsyncClient.builder()
+            .credentialsProvider(StaticCredentialsProvider.create(awsCreds))
+            .region(Region.of(vectorizeServiceParameters.get("region").toString()))
+            .build();
+    final CompletableFuture<InvokeModelResponse> invokeModelResponseCompletableFuture =
+        client.invokeModel(
+            request -> {
+              try {
+                String inputJson =
```

Comment:
Since we need UTF-8 bytes, instead use `writeValueAsBytes()` to avoid separate conversion (Jackson directly outputs encoded content avoid both String allocation and encoding step)